extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufReader.java

```diff
@@ -80,7 +80,41 @@ public OpenCensusProtobufReader(
   @Override
   public CloseableIterator<InputRow> read()
   {
-    return CloseableIterators.withEmptyBaggage(readAsList().iterator());
+    return new CloseableIterator<InputRow>()
+    {
+      private boolean closed;
```

Comment:
Follow up question. Does this iterator class need to be thread-safe?

Reply:
Nevermind. Based on what I see returned from `withEmptyBaggage` and `wrap` probably not